clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -137,31 +166,21 @@ CompletableFuture<ClientResponse> sendAutoCommit(final Map<TopicPartition, Offse
                     }
                     return null;
                 });
-        return future;
-    }
-
-    public void clientPoll(final long currentTimeMs) {
-        this.autoCommitState.ifPresent(t -> t.ack(currentTimeMs));
     }
 
-    // Visible for testing
-    Queue<StagedCommit> stagedCommits() {
-        return this.stagedCommits;
-    }
 
-    private class StagedCommit {
+    private class UnsentOffsetCommitRequest {
         private final Map<TopicPartition, OffsetAndMetadata> offsets;
         private final String groupId;
         private final GroupState.Generation generation;
         private final String groupInstanceId;
         private final NetworkClientDelegate.FutureCompletionHandler future;
 
-        public StagedCommit(final Map<TopicPartition, OffsetAndMetadata> offsets,
-                            final String groupId,
-                            final String groupInstanceId,
-                            final GroupState.Generation generation) {
+        public UnsentOffsetCommitRequest(final Map<TopicPartition, OffsetAndMetadata> offsets,
+                                         final String groupId,
+                                         final String groupInstanceId,
+                                         final GroupState.Generation generation) {
             this.offsets = offsets;
-            // if no callback is provided, DefaultOffsetCommitCallback will be used.
```

Comment:
I'm wondering in the new code, did we still trigger `DefaultOffsetCommitCallback` if user does not specify a callback? Currently it seems we always just use a plain `FutureCompletionHandler.future` without registering any callback. Is that a behavioral change?

Reply:
I'm moving the Callback trigger entirely to the foreground thread. As you can see, the `PrototypeAsyncConsumer` invokes: 
``` 
       future.whenComplete((r, t) -> {
            if (t != null) {
                callback.onComplete(offsets, new KafkaException(t));
            } else {
                callback.onComplete(offsets, null);
            }
        });
  ```

Reply:
I'll double check if the defaultOffsetCommitCallback is being invoked when user provides a null callback.

Reply:
I do not think we do: in case `commitAync()` is called we just passed in `null` as the cb. In that case we would actually have an NPE when triggering:

```
callback.onComplete(offsets, new RuntimeException(t));
```

right?

Reply:
Thanks, I'll add a test case for this.

Reply:
I believe this was addressed in the last revision. See PrototypeAsyncConsumer:

```
final OffsetCommitCallback commitCallback = callback == null ? new DefaultOffsetCommitCallback() : callback;
future.whenComplete((r, t) -> { ... } });
 ```

But i'll add a test for this.